### ISSUE TITLE
[BUG] Significant decrease in the select rate on clickhouse cluster after carbonapi update from 0.14.2 to 0.15.2

### ISSUE THREAD

**Issue by mpostument** [go-graphite/carbonapi#590]
**Describe the bug**
Significant decrease in the select rate on clickhouse cluster after carbonapi update from 0.14.2 to 0.15.2 (0.15.1)
Downgrade to 0.14.2 set select rate on the previous level. 

**CarbonAPI Version**
After update from 0.14.2 to 0.15.2

**CarbonAPI Configuration:**
```yaml
listen: "0.0.0.0:8082"
# Controls headers that would be passed to the backend
headersToPass:
  - "X-Dashboard-Id"
  - "X-Grafana-Org-Id"
  - "X-Panel-Id"
# Max concurrent requests to CarbonZipper
concurency: 20
cache:
   type: "mem"
   size_mb: 0
   defaultTimeoutSec: 5
# Amount of CPUs to use. 0 - unlimited
cpus: 0
# Timezone, default - local
tz: ""

maxBatchSize: 100
graphite:
    # Host:port where to send internal metrics
    # Empty = disabled
    host: "host:2003"
    interval: "10s"
    prefix: "prefix"
    pattern: "{prefix}.{fqdn}"
# Maximium idle connections to carbonzipper
idleConnections: 10
pidFile: ""
upstreams:
    # Number of 100ms buckets to track request distribution in. Used to build
    # 'carbon.zipper.hostname.requests_in_0ms_to_100ms' metric and friends.
    # Requests beyond the last bucket are logged as slow (default of 10 implies
    # "slow" is >1 second).
    # The last bucket is _not_ called 'requests_in_Xms_to_inf' on purpose, so
    # we can change our minds about how many buckets we want to have and have
    # their names remain consistent.
    buckets: 10
    timeouts:
        find: "10s"
        render: "50s"
        connect: "200ms"
    # Number of concurrent requests to any given backend - default is no limit.
    # If set, you likely want >= MaxIdleConnsPerHost
    concurrencyLimitPerServer: 1000

    # Configures how often keep alive packets will be sent out
    keepAliveInterval: "30s"

    # Control http.MaxIdleConnsPerHost. Large values can lead to more idle
    # connections on the backend servers which may bump into limits; tune with care.
    maxIdleConnsPerHost: 1000
    backendsv2:
        backends:
          -
            groupName: "clickhouse"
            protocol: "carbonapi_v2_pb"
            lbMethod: "roundrobin"
            maxTries: 5
            # amount of metrics per fetch request. Default: 0 - unlimited. If not specified, global will be used
            maxBatchSize: 0
            # interval for keep-alive http packets. If not specified, global will be used
            keepAliveInterval: "30s"
            # override for global concurrencyLimit.
            concurrencyLimit: 1000
            # override for global maxIdleConnsPerHost
            maxIdleConnsPerHost: 1000
            # per-group timeout override. If not specified, global will be used.
            # Please note that ONLY min(global, local) will be used.
            timeouts:
                # Maximum backend request time for find requests.
                find: "10s"
                # Maximum backend request time for render requests. This is total one and doesn't take into account in-flight requests.
                render: "50s"
                # Timeout to connect to the server
                connect: "200ms"
            servers: ["server"]
# If not zero, enabled cache for find requests
# This parameter controls when it will expire (in seconds)
# Default: 600 (10 minutes)
expireDelaySec: 10
logger:
    - logger: ""
      file: "/mnt/log/carbonapi/carbonapi.log"
      level: "warn"
      encoding: "json"
      encodingTime: "iso8601"
      encodingDuration: "seconds"
```

**Additional context**
Select rate after update
![alerts](https://user-images.githubusercontent.com/9252594/112512477-11eb9b00-8d9c-11eb-8b32-7a988ca0741e.png)

**Comment by deniszh** [comment]
hi @mpostument ,

but why do you considering that as a bad thing?

**Comment by mpostument** [comment]
I don't know if this bad, but this is not usual. And do not understand why this happening

**Comment by Civil** [comment]
There was a fix in 0.15.0 for cases when maxBatchSize on global level is not 0 (in your case it's 100), it was overriding maxBatchSize that was set on backend level (0 - unlimited in your case). So instead of splitting render requests by 100 metrics you now have one request for whatever url supplies, I think that's the reason why amount of selects decreased

**Comment by mpostument** [comment]
@Civil you were right. Thank  you
